### ISSUE TITLE
Redesign website Template

### ISSUE THREAD

**Issue by pragmaticivan** [hexpm/hex_web#188]
Suggestion for redesign of the layout of the website:

* [x] Layout base
* [x] Home
* [x] Packages List
* [x] Package Show

![hex2](https://cloud.githubusercontent.com/assets/301291/11324632/108356ca-90ec-11e5-9bc0-599f5b997bb6.png)

**Comment by pragmaticivan** [comment]
Packages Page.
![packageshex](https://cloud.githubusercontent.com/assets/301291/11324970/b69874ce-90f5-11e5-9d00-ee2da849b5d0.png)

**Comment by pragmaticivan** [comment]
Second Version for Packages.
![packagehex2](https://cloud.githubusercontent.com/assets/301291/11325412/db497a64-9101-11e5-8d46-5707fd5374cf.png)

**Comment by pragmaticivan** [comment]
Package View:

![hexpackageshow](https://cloud.githubusercontent.com/assets/301291/11330585/46edfdc0-9162-11e5-8ffc-0edcd87a0b06.jpg)

**Comment by pragmaticivan** [comment]
Package View:
![hexpackageshow](https://cloud.githubusercontent.com/assets/301291/11330591/5f80e082-9162-11e5-8980-caee620666f7.jpg)

**Comment by pragmaticivan** [comment]
cc @ericmj @josevalim 

Let me know if you guys have any suggestions, changes or new pages.

**Comment by josevalim** [comment]
Thanks Ivan! Quick feedback: we won't incorporate the Readme and it is
unlikely that we will have avatars for maintainers in the package page, so
you should consider how this would affect the design. :)

Great work! we should wait for @ericmj feedback too. He is the one deciding
everything.


-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by pragmaticivan** [comment]
Cool, I have some questions, I'm very new with Elixir, so I don't know all the specs related to the environment. Will the repo always be Github for packages? I was thinking about using The Gravatar image for that. And the info would come from Github. But Ok, let's wait @ericmj .

**Comment by ericmj** [comment]
Looks great Ivan. We might include the readme, but not right not at least because it would also require changes to the publication process and need to be stored in the database. We can't assume that the repo will always be from Github unfortunately, so I think we should just skip the Github customization.

The packages page is a little colorless, what do you think of making the package links red like on the front page? Not sure if you haven't changed it yet but the page selector seems out-of-place with the new design.

**Comment by pragmaticivan** [comment]
Even if not from Github, will aways be using git? Because I think we still can calculate the contributors and have the emails from the commits.

By the way any "new feature" can stay on the template, but I'll implement another example not showing the feature. So  in the future if you guys decide that we're going to implement this we will have the design for that.

**Comment by ferd** [comment]
I sometimes use mercurial and would sure like it if I could publish packages without switching away. It might give me a chance to stop being forced to use git for everything in the world because the moment I don't, I have to get into huge arguments about "ease of contribution" with people who don't plan to contribute anything anyway!

**Comment by josevalim** [comment]
We do have their emails, so that's enough for gravatar, but I am still not
a fan of the feature. :)

Also, we shouldn't have anything related to Github or git. How a package is
published should not depend on how it is developed. :)



-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by pragmaticivan** [comment]
There's also something missing. Should we have a link to hexdocs  in case the package has documentation?

**Comment by ericmj** [comment]
@pragmaticivan Yes, a general link and a link for each package version.

**Comment by pragmaticivan** [comment]
Package List Update:
![hexpackagelist](https://cloud.githubusercontent.com/assets/301291/11346960/3460305c-91d4-11e5-94df-e347ab15d237.jpg)

**Comment by pragmaticivan** [comment]
Updated Package View with some modifications.
![packageview2](https://cloud.githubusercontent.com/assets/301291/11347714/9a49b8cc-91d7-11e5-8ed4-4f03921a4dac.jpg)

**Comment by ericmj** [comment]
Looks great. Don't forget to change collaborators to "Maintainers" and a list of names instead of pictures.

**Comment by pragmaticivan** [comment]
What about this one?
![packageview2](https://cloud.githubusercontent.com/assets/301291/11381862/7ef265c4-92b3-11e5-97c8-331a504f192b.jpg)

**Comment by ericmj** [comment]
Beautiful!

**Comment by pragmaticivan** [comment]
So, full approved?

**Comment by pragmaticivan** [comment]
I think we can reuse the documentation pages and just change the title styles using the same as in package view.

**Comment by ericmj** [comment]
Yes, go ahead. We should also remove the Github issues and pull requests from the package page.

**Comment by josevalim** [comment]
I think the package page could be improved in the sense we have too much information packed (ha) on the right. The version currently online seems to spread the information better. I also don't like that the links come at the bottom. What if we kept only the Config and Stats on the right side? Or maybe only the config... if we can find a nice way to show the stats as you did for the home page.

Thanks a lot @pragmaticivan!

**Comment by pragmaticivan** [comment]
Cool, I'm going to do some UX Homework and try to come with something that solves this issue.

I also started some design for the menu as well

<img width="437" alt="screen shot 2015-11-24 at 4 04 52 pm" src="https://cloud.githubusercontent.com/assets/301291/11384518/4bf0a282-92c5-11e5-8a36-042b3b7dcd7b.png">

Do you think this menu fits good the template?

**Comment by josevalim** [comment]
It looks nice @pragmaticivan! 

I have one question though: on the home and search page, you used purple for heading and red for links. Should we follow the same conventions on the package page and menu? I am just wondering, I don't have any design expertise. :D

**Comment by pragmaticivan** [comment]
Actually I don't have design expertise, I usually don't do design for anything, so It's just a "hobby" haha. I need to work a little bit more on those kind of things.  I think in my mind at this time I was just following the colors of the new logo. Probably you have access to this new logo design.

**Comment by josevalim** [comment]
@pragmaticivan yes, I do have access to the new logo. :) I think we should stick to a set of colors, the new logo has 6 and it may be weird to use a different combination on every page.

**Comment by pragmaticivan** [comment]
Is there a way to have more than one License in a package? I didn't find a case like that browsing on the project. 

Should I use: 
`MIT license` or as on the current website: 'Licenses: MIT` ?

**Comment by ericmj** [comment]
Yes, a project can have multiple licenses. We should use the string that is in the package metadata, for MIT I think the string usually is just "MIT".

**Comment by pragmaticivan** [comment]
![package_show](https://cloud.githubusercontent.com/assets/301291/11416699/227c3d5c-93c5-11e5-93b6-945cccd7cc89.jpg)
Package Show Page.

**Comment by ericmj** [comment]
I think that looks great but build tools can be moved to the sidebar because it is less important than everything else. WDYT @josevalim?

**Comment by pragmaticivan** [comment]
<img width="655" alt="screen shot 2015-11-25 at 10 44 00 pm" src="https://cloud.githubusercontent.com/assets/301291/11416816/113f94ca-93c6-11e5-9425-54a32fea97f6.png">

Here's the final menu concept.

**Comment by pragmaticivan** [comment]
![package_show2](https://cloud.githubusercontent.com/assets/301291/11416902/cea6dc12-93c6-11e5-8555-471cf1f815ea.jpg)
Package Page with Build tools in the sidebar.

**Comment by josevalim** [comment]
It looks great! @pragmaticivan!

Does it make sense to move the stats to the left side and use icons similar to the one on the home page? 

Also, the right column is too high up, I think it should start after the Title+Description, similar to today.

Finally, there are still some inconsistencies between the package and the home page. I am not sure if we want to unify them further. I would try something like this:


```
Ecto v0.0.1
a package that does cool things


Maintainers                                [Config         ]
  ....
Links
  ...
License
  ...


      VERSIONS         DEPENDENCIES
----------------------------------------

```

In other words, it would be nice if the headings used in the home page where used throughout. :)

**Comment by josevalim** [comment]
Ah, I would also keep the package title red... because they are red on all other pages. :)

**Comment by pragmaticivan** [comment]
Cool.
- On the current site, the package title are blue but this one Should not be Red. It's not a Link.
- The headings are black because they're not Main headings on this page, I tested using purple but didn't look very good (I can show the tests).
- I think moving the States to the left would be good, I was trying to experiment some new approaches with graphs but didn't look very good. I'm going to try to do that.
- I'm gonna move the right column to align as you said, makes sense.

**Comment by josevalim** [comment]
Agreed on all points! Thanks for the clarifications.

-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by pragmaticivan** [comment]
![package_show3](https://cloud.githubusercontent.com/assets/301291/11430698/58087cd6-943f-11e5-8e33-2e191b1ffc16.jpg)

I'm not sure if this the Versions and Dependencies looks good, because there'is not a lot of information. But here's another layout for that.

**Comment by pragmaticivan** [comment]
Need to improve the size of the fonts but we can imagine how would be the page.

**Comment by josevalim** [comment]
Thank you! This looks much nicer!

A couple other suggestions we could try:

* the stats icons look oddly centralized. can we make the stats icons take
only the left column instead of both left and right columns? If that
doesn't work, can we simply remove the line above the icons and rely on
more whitespace to improve the content distribution?

* the purple headings on the right column still look odd to me, sorry.
:( Maybe we should keep them black as with Maintainers, Links and so on?

Regarding the Versions and Dependencies, I was thinking we could style them
the same we style the "Most Downloaded" on the home page. Or how we styled
the "getting started" heading. That's what I meant by reusing the home page
headings :) no idea if it would look good though or if you already tried it!



-- 


*José Valimwww.plataformatec.com.br
<http://www.plataformatec.com.br/>Founder and Director of R&D*

**Comment by pragmaticivan** [comment]
![package_show4](https://cloud.githubusercontent.com/assets/301291/11431678/df84f452-9451-11e5-9b2d-d30d0ef959fb.jpg)

I tried the style of Most Downloaded but didn't look very good.

**Comment by pragmaticivan** [comment]
What about this version?

**Comment by josevalim** [comment]
I like it! And what do *you* think? :D

Three final questions:

1. can we still increase a bit the vertical spacing (margin-top/bottom) in the stat items?
2. Should MIT License be in red? Afaik, it is not a link, just some random text
3. Should All Versions be a button? I have been told buttons should be used when causing an action from the server (like posting a form, deleting an item) and "All versions" is just navigation. I have no idea if this is meant to be true though

Thank you so much for the back and forth on this, I appreciate it!

**Comment by pragmaticivan** [comment]
> can we still increase a bit the vertical spacing (margin-top/bottom) in the stat items?

Yes, the spacing is something that I need to clean a little bit more. This kind of design spacing is something that I always change when I start coding the client-side of the project.

> Should MIT License be in red? Afaik, it is not a link, just some random text

I'm not sure if it should be a Link, Just added there in case. But we have styles on PSD for non-link. Should be easy to change in this case.

> Should All Versions be a button? I have been told buttons should be used when causing an action from the server (like posting a form, deleting an item) and "All versions" is just navigation. I have no idea if this is meant to be true though

Personally I don't like this approach of button as well. Even load more doesn't look good for me. 
Maybe Versions should be a page like in https://rubygems.org/gems/devise/versions. And instead we would use a link with `all versions` as well.

**Comment by josevalim** [comment]
Ah, I see. It is a button because it loads things in place. That sounds good to me then but I agree in the long term we can create a separate page!

**Comment by pragmaticivan** [comment]
![package_show5](https://cloud.githubusercontent.com/assets/301291/11438094/4e4182d4-94a7-11e5-87bb-c6f17d76413d.jpg)

The padding/marging space may change when start coding the front-end but that's the final result of the design.

**Comment by josevalim** [comment]
Beautiful, thank you!

**Comment by pragmaticivan** [comment]
So, should we close this task? or Still waiting for @ericmj ? I'm sending the PSD by email just in case.

**Comment by josevalim** [comment]
I got it, thanks! We should wait for @ericmj to close this. :)

**Comment by ericmj** [comment]
Closing this. If issues and/or questions come up during the psd => html conversion please open up a separate issue where we can track those. Thank you!
